File: apple/Sources/ReactNativeWatchOS/Modifiers/LabelIconToTitleSpacing.swift
import SwiftUI

/// SwiftUI `.labelIconToTitleSpacing(_:)` — sets the spacing between a `Label`'s
/// icon and its title.
///
/// LIMITATION: this API is only available on watchOS 26+ SDKs, newer than this
/// package's build floor (watchOS 9). Referencing the symbol — even behind
/// `#available` — would fail to compile on the supported build SDKs, so this is
/// a documented no-op: the view is returned unchanged. Registering the `$type`
/// keeps dispatch consistent and lets the no-op be swapped for a real
/// implementation once the SDK floor is raised.
enum RNWLabelIconToTitleSpacingModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("labelIconToTitleSpacing") { view, _, _ in view }
    }
}

File: src/modifiers/labelIconToTitleSpacing.ts
import { createModifier } from './createModifier';

/// SwiftUI `.labelIconToTitleSpacing(_:)`. Sets the spacing between a `Label`'s
/// icon and its title.
///
/// LIMITATION: `.labelIconToTitleSpacing(_:)` only exists on watchOS 26+ SDKs,
/// newer than this package's build floor. To keep the native side compiling on
/// every supported SDK it is currently a documented no-op — the view is
/// returned unchanged. The factory ships so callers can adopt it without churn
/// once the native side raises its SDK and wires it up.
export function labelIconToTitleSpacing(spacing: number) {
  return createModifier('labelIconToTitleSpacing', { spacing });
}
